frontend/app/mocks/node.ts

```diff
@@ -72,6 +72,51 @@ function getPreferredLanguageEntity(id: string | readonly string[]) {
   return parsedPreferredLanguage;
 }
 
+/**
+ * Retrieves list of letter entities based on the provided user ID.
+ *
+ * @param userId - The user ID to look up in the database.
+ * @returns The letter entity if found, otherwise throws a 404 error.
+ */
+function getLetterEntities(userId: string | readonly string[]) {
+  const parsedUserId = z.string().uuid().safeParse(userId);
+  const letterEntity = !parsedUserId.success
+    ? undefined
+    : db.letter.findMany({
+        where: {
+          userId: {
+            equals: parsedUserId.data,
+          },
+        },
+      });
+  if (!letterEntity) {
+    throw new HttpResponse('Letter Not found', { status: 404, headers: { 'Content-Type': 'text/plain' } });
+  }
+
+  return letterEntity;
+}
+
+/**
+ * Retrieves a PDF entity based on the provided referenceId ID.
+ *
+ * @param id - The reference Id to look up in the database.
+ * @returns The PDF entity if found, otherwise throws a 404 error.
+ */
+function getPdfEntity(referenceId: string | readonly string[]) {
```

Comment:
Why can `userId` be an array of strings here? I feel like it makes sense to limit this to just a single user id.